staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1275,6 +1275,7 @@ func RunTestList(ctx context.Context, t *testing.T, store storage.Interface, com
 			name:        "test List with resource version after writes, match=Exact",
 			prefix:      "/pods/",
 			pred:        storage.Everything,
+			skip:        watchCacheEnabled && !supportsConsistenRead,
```

Comment:
hrmm, I'm surprised to see us adding new skips for test cases for features that are on by default ... does this mean ConsistentListFromCache breaks this test case? did we know that when we enabled it by default?

same comment for the new skip below

Reply:
Without consistent read we cannot serve MatchExact on revision that was not observed by cache. This is expected edge case for a test I just added in the effort. 

>  I'm surprised to see us adding new skips for test cases for features that are on by default

I wanted to test all combinations configurable by user.

Reply:
> This is expected edge case for a test I just added in the effort.

Ah, ok, I didn't know if this test predated consistent watch cache and we had disabled to feature to keep the test passing

Reply:
It didn't predate consistent reads, it was added for testing snapshots in KEP-4988, however I missed that the scenario depends consistent read feature flag. It might not be enabled in watch cache.

The test passes because `match=Exact` are currently always passed to etcd which always handles consistent reads correctly. However when we introduce the serving `match=Exact` from cache, we will need this consistent read from cache to be enabled.

Reply:
hmm... but... if a request comes in with match=Exact which the watch cache cannot honor, what does it to? Pass through to etcd, or respond incorrectly, or error?

Reply:
Agree on trying to get the behavior conformance tests passing independent of feature flags, skipping based on feature flags feels like it's not quite accomplishing that

Reply:
> what does it do? Pass through to etcd, or respond incorrectly, or error? If it passes through in that case, shouldn't the test still pass?

Currently it passes it to etcd and the test will pass. The reason to fix it to prepare for the following PR that will enable serving requests from cache. We will add more tests to cover a new feature flag 

https://github.com/kubernetes/kubernetes/pull/130423/commits/881d45c2a9d6137edada6f48891c8e8f2f57da6c#diff-d82df65cc0625a3f621333245e76fbc801b579eedd3d266fe36eeb99366b5680

Reply:
> Currently it passes it to etcd and the test will pass.

So... shouldn't it continue passing with all combinations of feature flags, either because they support the request or they pass it through?